app/components/agenda/agendaitem/agendaitem-decision.hbs

```diff
@@ -1,69 +1,183 @@
                 @isEditable={{user-may "manage-decisions"}}
-                @onEdit={{this.toggleEdit}}
+                @onEdit={{this.toggleEditPill}}
               />
             </div>
           {{else}}
             <Agenda::Agendaitem::AgendaitemDecisionEdit
               @decisionActivity={{@decisionActivity}}
               @onSave={{perform this.updateAgendaitemPiecesAccessLevels}}
-              @onCancel={{this.toggleEdit}}
+              @onCancel={{this.toggleEditPill}}
             />
           {{/if}}
         </Group.Item>
       </Toolbar.Group>
     </Auk::Toolbar>
-  </Auk::Panel::Header>
-  <Auk::Panel::Body @noPadding={{true}}>
-    {{#if this.loadReport.isIdle}}
-      {{#if this.report}}
-        <Documents::DocumentCard
-          @piece={{this.report}}
-          @isEditable={{true}}
-          @agendaContext={{@agendaContext}}
-          @hideNewerVersions={{true}}
-          @onOpenUploadModal={{@onOpenUploadModal}}
-          @onAddPiece={{this.attachNewReportVersion}}
-          @didDeleteContainer={{perform this.loadReport}}
-          @bordered={{false}}
+  </Auk::Panel::Body>
+
+  {{#if this.loadReport.isIdle}}
+    {{#if this.isEditing}}
+      <Auk::Panel::Body
+        class="au-o-flow {{if this.isEditing 'au-u-background-gray-100'}}"
+      >
+        <h4 class="auk-h4">Betreft</h4>
+        <Agenda::Agendaitem::Decision::Editor
+          @handleRdfaEditorInit={{this.handleRdfaEditorInitBetreft}}
+          @piecePart={{this.betreftPiecePart}}
+          @onPiecePartVersionClicked={{this.setBetreftEditorContent}}
+          @onUpdateContentClicked={{this.updateBetreftContent}}
         />
-      {{else if (user-may "manage-decisions")}}
-        <div class="auk-u-p-2">
-          <AuButton
-            data-test-agendaitem-decision-upload-file
-            @skin="naked"
-            @icon="upload"
-            {{on "click" this.toggleIsAddingReport}}
-          >
-            {{t "signed-decision-add"}}
-          </AuButton>
-        </div>
-      {{/if}}
+        <h4 class="auk-h4 au-u-padding-top">Beslissing</h4>
+        <Agenda::Agendaitem::Decision::Editor
+          @handleRdfaEditorInit={{this.handleRdfaEditorInitBeslissing}}
+          @piecePart={{this.beslissingPiecePart}}
+          @onPiecePartVersionClicked={{this.setBeslissingEditorContent}}
+          @onUpdateContentClicked={{this.updateBeslissingContent}}
+          @disableUpdateContent={{not this.nota}}
+        />
+      </Auk::Panel::Body>
     {{else}}
-      <Auk::Loader />
+      {{#if (and this.report.file (not this.report.pieceParts))}}
+        <Auk::Panel::Body class="au-o-flow" @noPadding={{true}}>
+          <Documents::DocumentCard
+            @piece={{this.report}}
+            @isEditable={{true}}
+            @agendaContext={{@agendaContext}}
+            @hideNewerVersions={{true}}
+            @onOpenUploadModal={{@onOpenUploadModal}}
+            @onAddPiece={{this.attachNewReportVersionDepr}}
```

Comment:
`@onAddPiece` will pass a `piece` record to `this.attachNewReportVersionDepr`. We actually need a `report` record, otherwise ED cries. You'll need to find a way to cast it/convert it/create a new report with the same attributes as the piece and delete the piece.

Not sure if there's a cleaner way, maybe parametrize the document-card so you can tell which kind of piece to create.